### ISSUE TITLE
Remove fastqc from pipeline, replace %>=Q30 calculation with faster source

### ISSUE THREAD

**Issue by AmandaBirmingham** [ucsd-ccbb/covid_sequencing_analysis_pipeline#89]
<img width="611" alt="Screen Shot 2021-03-31 at 4 01 35 PM" src="https://user-images.githubusercontent.com/10677935/113221715-72665680-923a-11eb-9aff-c1542f755055.png">

@niemasd  suggests https://github.com/dayedepps/q30

**Comment by AmandaBirmingham** [comment]
@artnasamran determined that the https://github.com/dayedepps/q30 code does not calculate the same metric as fastqc: fastqc calculates the percent of READS >=Q30 while the dayedepps code calculates the percent of BASES >=Q30.  He is investigating whether there is another easy-to-incorporate tool that will give us a metric equivalent to the fastqc one; if not, we will stop and get prioritization on how important this exact metric is ...

**Comment by niemasd** [comment]
@AmandaBirmingham Can you clarify what exactly it means for a read to be >= Q30? Does it mean that every single base in the read is >= Q30? Or something similar?

**Comment by niemasd** [comment]
According to [this link](https://bioinf.comav.upv.es/courses/sequence_analysis/read_cleaning.html):

> Q30 is the percentage of **bases** in the reads with a phred quality equal or bigger than 30.

Figure 2 of [this Illumina document](https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf) counts total **bases** >= Q30

According to [this link](https://horizondiscovery.com/en/blog/the-5-ngs-qc-metrics-you-should-know):

> %Q30: The percentage of **bases** with a quality score of 30 or higher, respectively

According to [this ThermoFisher blog post](https://www.thermofisher.com/blog/behindthebench/what-is-accuracy-in-next-generation-sequencing-ngs/):

> You are  likely have encountered a metric called %bases greater than or equal to Q30 which translates to the percent of sequenced **bases** that have a predicted quality score of 30 or more.

According to Table 1 of [this paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6487961/):

> %≥Q30: The percentage of **bases** having a quality score of 30 or higher

And I couldn't find any resources that said anything about Q30 with respect to proportion/percentage of reads

**TL;DR:** I'm fairly certain that the ">=Q30" is referring to the percentage of **bases** (not reads) that have a quality score of at least Q30

**Comment by niemasd** [comment]
Found [this](https://github.com/s-andrews/FastQC/pull/42/commits/499aa91c6454651ceb88ab17e2f751477470e522) in the FastQC code (the precise moment when they added Q20 and Q30 metrics). The [`q20_count`](https://github.com/s-andrews/FastQC/pull/42/commits/499aa91c6454651ceb88ab17e2f751477470e522#diff-4f1784309925c9fe5cdcb6c36c63d5c27e2ffd88cb4ac24bb8051b2118195d8cR48) and [`q30_count`](https://github.com/s-andrews/FastQC/pull/42/commits/499aa91c6454651ceb88ab17e2f751477470e522#diff-4f1784309925c9fe5cdcb6c36c63d5c27e2ffd88cb4ac24bb8051b2118195d8cR49) variables are global to the [`BasicStats`](https://github.com/s-andrews/FastQC/pull/42/commits/499aa91c6454651ceb88ab17e2f751477470e522#diff-4f1784309925c9fe5cdcb6c36c63d5c27e2ffd88cb4ac24bb8051b2118195d8cR37) class, and [those global variables are incremented in each base of each sequence without ever being reset as far as I can tell](https://github.com/s-andrews/FastQC/pull/42/commits/499aa91c6454651ceb88ab17e2f751477470e522#diff-4f1784309925c9fe5cdcb6c36c63d5c27e2ffd88cb4ac24bb8051b2118195d8cR137), which suggests to me that, even in FastQC, these are indeed per **base** rather than per read

**Comment by artnasamran** [comment]
@niemasd @AmandaBirmingham Do you know whether the stakeholders want %>=Q30 bases or average %>=Q30 sequences?

Our original code was calculating % >=Q30 from the "#Quality Count" table in the standard fastqc_data.txt output from fastqc. I was calculating this by summing counts of Q>=30 and dividing them by the total counts.  Here's an example excerpt of a table from fastqc_data.txt: 
>>Per sequence quality scores	pass
#Quality	Count
16	2.0
17	12.0
18	43.0
19	117.0
20	249.0
21	541.0
22	1243.0
23	2464.0
24	4739.0
25	8456.0
26	13710.0
27	21038.0
28	29277.0
29	39743.0
30	52796.0
31	71578.0
32	103956.0
33	180065.0
34	461070.0
35	1449713.0
>>END_MODULE

We've been reporting average %>=Q30 by sequence.

If they want %>=Q30 by base, we can use q30.py from https://github.com/dayedepps/q30 as is. Otherwise, it's a quick fix to calculate the average quality per sequence.

**Comment by rob-knight** [comment]
This stakeholder at least has never thought about this issue. However, given that fastqc was proposed as an acceptable solution previously, it would be a good idea to match what it does if that is technically feasible/fast. If some other solution is planned, it would be great to see a correlation between old vs new results — if it’s very close, which it probably is, I think the team would prefer speed over one particular definition.
